dasharo-compatibility/dcu.robot

```diff
@@ -0,0 +1,160 @@
+*** Settings ***
+Library             Collections
+Library             OperatingSystem
+Library             Process
+Library             String
+Library             Telnet    timeout=30 seconds    connection_timeout=120 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+# TODO: maybe have a single file to include if we need to include the same
+# stuff in all test cases
+Resource            ../sonoff-rest-api/sonoff-api.robot
+Resource            ../rtectrl-rest-api/rtectrl.robot
+Resource            ../variables.robot
+Resource            ../keywords.robot
+Resource            ../keys.robot
+Resource            ../pikvm-rest-api/pikvm_comm.robot
+
+# TODO:
+# - document which setup/teardown keywords to use and what are they doing
+# - go threough them and make sure they are doing what the name suggest (not
+# exactly the case right now)
+Suite Setup         Run Keywords
+...                     Prepare Test Suite
+# ...    AND
+# ...    Prepare DCU Test Environment
+# ...    AND
+# ...    Make Sure That Flash Locks Are Disabled
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+
+
+*** Test Cases ***
+DCU001.001 Change the UUID
+    [Documentation]    This test case verifies that the UUID encoded in the DMI
+    ...    table of an image can be changed using DCU.
+    Skip If    not ${UUID_SETTABLE}    DCU001.001 not supported
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    DCU001.001 not supported
+    Skip
+    ${uuid}=    Set Variable    96bcfa1a-42b4-6717-a44c-d8bbc18cbea4
+    ${result}=    Run Process    bash    -c    ./dcu/dcu smbios -u ${uuid} ./dcu/coreboot.rom
```

Comment:
This assumes I need to have dcu in local directory? What happens if I do not? How is it documented?

Reply:
What is coreboot.rom and what should I put there.

In other tests, we are using fw_file variable IIRC, which is more widely used already.

Reply:
DCU will always get downloaded locally soon as the test runs. Reworked it to use the `${FW_FILE}` as coreboot.rom instead of extra downloading [here](https://github.com/Dasharo/open-source-firmware-validation/pull/139/commits/ad98d1323929b34a1e69ec7c1e73bc6ecd7bed42#diff-7ee5e21e59cd627383812396fcf45a6e7ecb3e2f096ab5ae5203a0c6bd75989fR156)